### DISCUSSION TITLE
async-data gives errors when using wire:model

### DISCUSSION THREAD

**Discussion by markgoedegebure** [wireui/wireui#322] [Q&A]
**Describe the bug**
When using async-data option in x-select it works as expected, untill i add a wire:model. At that point i get alot of errors in the console.

**To Reproduce**
1. Add an x-select with async-data option and wire:model option

**Expected behavior**
The model defined in wire:model contains the value of the selected option without any errors

**Screenshots**
![image](https://user-images.githubusercontent.com/44771394/174049219-2bb58790-1710-4a80-9431-d6ab378fee85.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version: 102.0.5005.115

**Additional context**
Laravel v9.17.0
Alpine v3.10.2
Livewire v2.10.5
WireUI v1.6

**Comment by markgoedegebure** [comment]
the problem seems to occur in all wire:model in all selects, with or without async-data option...

**Comment by ph7jack** [comment]
@markgoedegebure Share your component code and the wire model value in the PHP class

**Comment by markgoedegebure** [comment]
Hello @PH7-Jack ,
Thanks for your reply.

My component code:

`
<x-select
                        wire:model.defer="user"
                        :async-data="[
                                'api' => route('api.users.index'),
                                'method' => 'POST',
                                'params' => [] 
                                ]"
                                option-label="name"
                                option-value="id"
                                label="Gebruikers"
                                :wire:key="'select'.now()"
                    />`

My Livewire class:
`
class UserForm extends Component
{
    public User $user;

    public function mount(User $user)
    {
            $this->fill($this->user);
    }

    public function render()
    {
        return view('livewire.users.user-form');
    }
}
`

**Comment by ph7jack** [comment]
The wire:model cannot handle the whole model instance, you must use only a specific column
`<x-select wire:model.defer="user.id"`

**Comment by markgoedegebure** [comment]
That doesn't seem to do the trick. Also the same problem appears in a simple select with an array as values:
`<x-select
                        wire:model="status"
                        :options="['Active', 'Inactive']"
                        label="Status"
                    />`
When i delete the line `wire:model="status"`, the selectbox is properly showing, but when i add it, alot  errors appear.

These are the errors that appear following the list of warnings above:
![image](https://user-images.githubusercontent.com/44771394/174236786-fb8d69ca-feaa-450c-8182-afa987ad06d9.png)

**Reply by t1m0t** [reply]
@markgoedegebure Have you resolved it? Just in case, as I encountered this issue, I wrote this post: https://github.com/wireui/wireui/discussions/469.

**Comment by affektde** [comment]
We have also problems with Chrome on Mac OS X. Some team members have issues with all Selects with async-data options. I don't have a Mac, so we I need to get more console information about it. But something is weird with Mac OS X Chrome Version.

**Reply by t1m0t** [reply]
@affektde Have opened a discussion/issue for this?

**Comment by ph7jack** [comment]
@markgoedegebure Try removing it `:wire:key="'select'.now()"`, I'm not sure why you need to reset the component on every refresh, keep the component
